### ISSUE TITLE
Hitting Tab in console escapes macros

### ISSUE THREAD

**Issue by qymspace** [ranger/ranger#1460]
#### Runtime Environment

- Operating system and version: Ubuntu 16.04
- Terminal emulator and version: Gnome terminal
- Python version: 2.7.12
- Ranger version/commit:  Not Sure how to get this
- Locale: 

#### Current Behavior
Typing :shell -f mv %s  and hitting tab results in 
shell -f mv %%s destination_directory

#### Expected Behavior
shell -f mv %s destination_directory
Note that %s shouldn't be escaped
#### Context
Copy and move selections

**Comment by qymspace** [comment]
running ```apt install ranger``` on ubuntu says
```ranger is already the newest version (1.7.1-1).```
How to update from repository directly

**Comment by toonn** [comment]
Ah, indeed it does, but only if you include a ` ` after the `%s`. Thanks for the report!

**Comment by qymspace** [comment]
Yes. Unfortunately I have to include the ``` ``` to use the command. 
But thanks for the very impressive tool
I am using ```map mvto  eval fm.open_console('shell -f mv %s ')``` to open console to specify a folder to move selection to

**Comment by SBanya** [comment]
Hey everyone,

I spoke to Wojciech over email and I'd love to help out with some user tickets like these.

I'll try to replicate this issue with a related VM this weekend and will report if I saw this same issue or not.

**Comment by SBanya** [comment]
Hello,

After downloading that version of Ubuntu 16.04 Desktop 64 bit ISO, and running it in a VM with Ranger, I was also unable to replicate this tab completion issue.

Thank you for letting us know about what version of Ranger you're running.

Please let us know if you are still experiencing this issue, or how we can further assist you.

**Comment by toonn** [comment]
@SBanya, did you catch my comment? Because I *was* able to reproduce this.

**Comment by SBanya** [comment]
Hmm well I ran Ubuntu 16.04, ran Gnome terminal, installed Ranger, and tried doing tab completion with that command but no dice.

I probably had the latest version of Ranger as well since I downloaded it through sudo apt install.

How were you able to replicate this?

**Comment by SBanya** [comment]
toonn: Gotcha, would you like me to then redownload this VM, make sure that Ranger is the most updated version, and then reconfirm the same action?

If not, now that we've confirmed it on our end, what part of our codebase should we look into then? Is there any section regarding tab completion in particular?

I ask, because I did try to do the exact same style of tab completion like the initial user mentioned as well, so I understand what you mean regarding being specific about it.

**Comment by qymspace** [comment]
Yes. I am positive that this issue still exist. 
Are you hitting tab after a space. I.e  ```shell -f mv %s[SPACE][TAB TAB TAB TAB]```

What is your result?

**Comment by toonn** [comment]
@SBanya, as a general rule you never need to get a VM for reproduction *unless* you can't reproduce the issue. First you try reproducing it on the master branch version of ranger then the version the user reported. If it's a regression you `git bisect` between the know-working and know-broken versions. If at this point the issue's not fixed and you haven't been able to reproduce, then we go the VM route.

I'm not familiar with the code for tab completion tbh. `ranger/api/commands.py` seems a promising place to start looking though.

**Comment by SBanya** [comment]
@toonn
Gotcha, makes sense. Haven't used git bisect before so I'll read up on it, before attempting it with the Ranger Github repo.

Thanks for providing the relevant link to tab completion, I'll check it out.

**Comment by SBanya** [comment]
Would you do a git fork first before doing git bisect?

**Comment by toonn** [comment]
@SBanya it's not necessary. It's useful only if/when you want to contribute fixes.
